common/types/pb/pb.go

```diff
@@ -121,7 +121,12 @@ func CollectFileDescriptorSet(message proto.Message) (*descpb.FileDescriptorSet,
 	fdMap := map[string]*descpb.FileDescriptorProto{}
 	fd, _ := descriptor.ForMessage(message.(descriptor.Message))
 	fdMap[fd.GetName()] = fd
-	deps := fd.GetDependency()
+	// Initialize list of dependencies
```

Comment:
Hmm, nevermind, even though slices are typically copy on write, there's still a chance this would modify the underlying storage. 

Let's try:
```go
fileDeps := fd.GetDependency()
deps := make([]string, len(fileDeps), len(fileDeps))
copy(deps, fileDeps)
```

Reply:
I've created PR #324 to address what I think are the remainder of the potentially (inadvertent) mutable `append` calls on slices.

Reply:
More than a "chance" the original suggestion definitely shares storage with the slice returned by `fd.GetDependency()`, but there's a chance that this slice is already at its capacity.  Switched to the more idiomatic copy.